ckanext/qgov/common/anti_csrf.py

```diff
@@ -134,8 +135,8 @@ def validate_token(token):
     if 'hash' not in token_values:
         return False
 
-    expected_hmac = unicode(get_digest(token_values['message']))
-    if not hmac.compare_digest(expected_hmac, token_values['hash']):
+    expected_hmac = six.ensure_text(get_digest(token_values['message']))
```

Comment:
is it utf-8 under the hood?

Reply:
In Python 2, it's equivalent to calling the `unicode` function. Documentation suggests that UTF-8 is the default encoding (https://docs.python.org/2/library/functions.html#unicode says `unicode` behaves similarly to `str` except returning a Unicode object, https://www.programiz.com/python-programming/methods/built-in/str says that the default encoding of `str` is UTF-8).